---
title: 'COMPMUT Experiments 7: Presentation and analysis of growth curves with candidate cost genes'
author: "jpjh"
date: "compiled Feb 2021, edited Jul 2021"
output: rmarkdown::github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
rm(list=ls())

library(tidyverse)
library(ggplot2)
library(knitr)
library(forcats)

source("../functions/theme_pub.r")
theme_github <- function(x){
  theme_pub(base_size=12)
}

theme_set(theme_github())
```

## Growth rate experiments with strains overexpressing candidate cost genes

#### Experimental design

Genes were identified from the RNAseq data as those whose product might directly cause a fitness cost. To test this, these putatitve 'candidate cost' genes were amplified from the chromosome and cloned into the expression vector pME6032. pME6032 differs from pUCP18xK in that it also carries a *lacI^q^* gene, which causes repression of insert expression under normal growth conditions, and enables inducible expression with the addition of 100 µg/ml IPTG. The generated expression plasmids were miniprepped and sequenced to ensure a correct insert, and transformed into *P. fluorescens* SBW25 in triplicate. Empty vector was also transformed into *P. fluorescens* in triplicate as a negative control. These transformants are transformant replicates, `t_rep`. Each `t_rep` strain was tested in triplicate (`exp_rep`).

Overnight cultures grown in the presence of tetracycline (100 µg/ml) to select for pME6032 maintenance were subcultured into 150 µl KB + tetracycline ± 100 µM IPTG, and grown in the automatic plate reader with readings taken every 15 minutes. Several different plates were set up to test all of the lines. Raw OD600 readings were obtained and corrected by subtraction of the 'no inoculation' blank control. 

#### Plotting of growthcurve data

Load up data.

```{r}
gr1 <- read.csv("../data/COMPMUT_exp_data_7.csv", header=TRUE, sep=",") %>%
  mutate(pME6032 = gsub("PFLU_","PFLU",pME6032))
```

Calculate means for each set of technical replicates. Plot mean and an envelope of SE/CI for each transformant.

```{r}
gr1_summ <- gr1 %>% mutate(IPTG = factor(IPTG, levels=c("0","100"), labels=c("uninduced","induced"))) %>%
  group_by(cycle, t_rep, pME6032, IPTG) %>%
  summarise(mean = mean(OD600_corr), 
            n = n(), 
            se = sd(OD600_corr)/n, 
            ci = (qt(0.95/2 + 0.5, n-1)) * se) %>%
  rename(OD600_corr=mean)

(plot_fig16 <- ggplot(data=gr1_summ,
       aes(x=(cycle/4), y=OD600_corr,
           colour=IPTG,
           group=interaction(t_rep,pME6032,IPTG))) +
  geom_ribbon(aes(ymin=OD600_corr-se, ymax=OD600_corr+se, fill=IPTG),
              alpha=0.3, colour=NA) +
  geom_line() +
  scale_colour_manual(values=c("#53BCC2","#E98076")) +
  scale_fill_manual(values=c("#53BCC2","#E98076")) +
  labs(x="time (h)", y=expression(paste("OD600"["corrected"])), fill="", colour="") +
  scale_x_continuous(breaks=c(0,18,36)) +
  facet_grid(~pME6032) +
  theme(legend.position="bottom"))
```

The plots may need to be adjusted for visibility.

Output as `.svg`.

```{r}
svglite::svglite(height=1.5, width=7.2, file = "../plots/Fig16.svg")
plot_fig16 + theme_pub() + theme(legend.position="bottom")
dev.off()
```


#### Analysis

Preliminary attempts to fit curves to these data were unsuccessful — particularly bad at fitting to the PFLU_1169 induced curves, which are the key features to capture.

Therefore, summarise curves by calculating area under the curve (AUC), and using this in linear models.

As we have datapoints for all samples at all timepoints, AUC = sum of densities. 

Calculate AUC for all experimental replicates separately.

```{r}
gr1_AUC <- gr1 %>% mutate(IPTG = factor(IPTG, levels=c("0","100"), labels=c("uninduced","induced"))) %>%
  group_by(exp, well, exp_rep, t_rep, pME6032, IPTG) %>%
  summarise(AUC = sum(OD600_corr))
```

Plot to check it looks sensible.

```{r}
pd <- position_dodge(width=0.3)

gr1_AUC %>% ggplot(aes(x=pME6032, y=AUC, colour=IPTG, shape=t_rep)) +
  geom_point(position=pd) +
  theme(axis.text.x = element_text(angle=45, hjust=1))
```

Looks good.

We are interested in the ratio between growth when induced vs. growth when uninduced. 

To do this, arrange the `gr1_AUC` dataframe such that consecutive rows have ±IPTG, then use the `mutate()` and `lag()` functions to get the ratio between rows. 

```{r}
gr1_ratio <- gr1_AUC %>% ungroup() %>% arrange(exp, t_rep, exp_rep, pME6032, IPTG) %>%
  mutate(ratio = AUC / lag(AUC)) %>% filter(IPTG=="induced")
```

Again, plot to check.

```{r}
gr1_ratio %>% ggplot(aes(x=pME6032, y=ratio, shape=t_rep)) +
  geom_hline(yintercept=1, linetype="dotted") +
  geom_point(position=pd) +
  theme(axis.text.x = element_text(angle=45, hjust=1))
```

Looks good.

We are interested in whether this ratio varies between pME6032 inserts.

Analyse as an LMM.

```{r}
gr1_ratio <- gr1_ratio %>% mutate(clone = factor(paste(t_rep, pME6032)))

library(nlme)

mod11_1 <- gls(data=gr1_ratio, ratio ~ pME6032, method="REML")
mod11_2 <- lme(data=gr1_ratio, ratio ~ pME6032, 
              random = ~1|clone, method="REML")

anova(mod11_1, mod11_2)
```

The transformant replicate has a very strong (random) effect.

```{r}
data.frame(intercept = ranef(mod11_2), 
           clone = rownames(ranef(mod11_2))) %>% 
  tibble %>%
  arrange(-abs(X.Intercept.)) %>% select(clone, X.Intercept.) %>% kable()
```

And the very strongest effects are in PFLU_1169.

Generate plots for model validation.

```{r}
gr1_ratio <- gr1_ratio %>% mutate(mod11_2_resid  = resid(mod11_2, type = "normalized"),
              mod11_2_fitted = fitted(mod11_2))

ggplot(data=gr1_ratio, aes(x=mod11_2_fitted, y=mod11_2_resid)) + 
  geom_point() + labs(x="fitted values", y="residuals")

ggplot(data=gr1_ratio, aes(x=pME6032, y=mod11_2_resid)) + 
  geom_boxplot() + labs(x="pME6032 variant", y="residuals")

ggplot(data=gr1_ratio, aes(x=pME6032, y=mod11_2_resid)) + 
  geom_point(position="jitter") + labs(x="pME6032 variant", y="residuals")

ggplot(data=gr1_ratio, aes(sample=mod11_2_resid)) + stat_qq() + stat_qq_line()
```

The QQ plot looks like it might be significant.

```{r}
shapiro.test(resid(mod11_2, type="normalized"))
```

Borderline (0.06), suggesting that this model is valid.

Proceed with model reduction.

```{r}
mod11_2_ml <- update(mod11_2, method="ML")

mod11_3_ml <- update(mod11_2_ml, .~.-pME6032)

anova(mod11_2_ml, mod11_3_ml)
```

Highly significant -- the insert has a significant effect on the consequences of induction.

Perform post-hoc comparisons with the empty vector control (Dunnett's test).

```{r}
library(emmeans)

posthoc <- lsmeans(mod11_2, trt.vs.ctrl ~ pME6032, adjust="mvt")

contr <- data.frame(posthoc$contrasts) %>% mutate(sign = ifelse(p.value<0.05, "*", ""))

kable(contr)
```

Clear significant effect for PFLU_1169. More marginal effect for PFLU_1170. Consistent with the figure.

---

**[Back to index.](COMPMUT_index.md)**
