<h1 id="readme-for-data-files-associated-with-plasmid-fitness-costs-are-caused-by-specific-genetic-conflicts-by-hall-wright-harrison-muddiman-wood-paterson-and-brockhurst">Readme for: <em>Data files associated with “Plasmid fitness costs are caused by specific genetic conflicts” by Hall, Wright, Harrison, Muddiman, Wood, Paterson, and Brockhurst</em></h1>
<p>April 2021</p>
<p>Full analysis scripts associated with this raw data can be found at <a href="github.com/jpjh/COMPMUT">github.com/jpjh/COMPMUT</a>.</p>
<h3 id="experimental-measurements-of-relative-fitness-plasmid-cost-etc">Experimental measurements of relative fitness, plasmid cost, etc</h3>
<ul>
<li><a href="COMPMUT_exp_data_1.csv"><code>COMPMUT_exp_data_1.csv</code></a>. Competition data from knockout strains. Columns indicate experimental replicate (<code>experiment</code>), transconjugant replicate (<code>replicate</code>), host genotype (<code>host</code>), plasmid variant (<code>plasmid</code>), whether measurement was from the start or the end of the competition (<code>timepoint</code>), dilution factor (<code>dilution</code>), volume spread in µl (<code>spread</code>) and the counts of white and blue colonies on that plate (<code>count_white</code>, <code>count_blue</code>).</li>
<li><a href="COMPMUT_exp_data_2.csv"><code>COMPMUT_exp_data_2.csv</code></a>. Competition data from plasmid mutants. Columns are the same as for <code>COMPMUT_exp_data_1.csv</code>, except <code>marker</code> indicates the antibiotic marker used in the plasmid-bearing strain. Gm = gentamicin-resistant, Sm = streptomycin resistant.</li>
<li><a href="COMPMUT_exp_data_3.csv"><code>COMPMUT_exp_data_3.csv</code></a>. Competition data from knockouts and plasmid mutants. Columns are the same as <code>COMPMUT_exp_data_1.csv</code>.</li>
<li><a href="COMPMUT_exp_data_3-1.csv"><code>COMPMUT_exp_data_3-1.csv</code></a>. Conjugation data from knockouts and plasmid mutants. Columns are the same as from <code>COMPMUT_exp_data_1.csv</code> except <code>recipient</code> refers to the species of the recipient (Pf = <em>P. fluorescens</em>, Pp = <em>P. putida</em>), and <code>timepoint</code> explains not only whether the plate contained start or end counts, but also what it was selecting for (DR = donors and recipients, T = transconjugants).</li>
<li><a href="COMPMUT_exp_data_4.csv"><code>COMPMUT_exp_data_4.csv</code></a>. Competition data from expressing PQBR57_0059 variants <em>in trans</em>. Columns are the same as <code>COMPMUT_exp_data_1.csv</code> , except <code>exp_rep</code> refers to experimental replicate and <code>tc_rep</code> refers to transconjugant replicate. <code>plasmid_pucp</code> refers to the PQBR57_0059 allele on the pUCP18 variant, whereas <code>plasmid_pq</code> refers to the pQBR57 variant.</li>
<li><a href="COMPMUT_exp_data_5.csv"><code>COMPMUT_exp_data_5.csv</code></a>. Competition data from expressing pQBR57 <em>par</em> genes <em>in trans</em> alongside pQBR57. Columns are the same as <code>COMPMUT_exp_data_4.csv</code> . <a href="COMPMUT_exp_data_5-1.csv"><code>COMPMUT_exp_data_5-1.csv</code></a> is similar but from expressing alongside pQBR103.</li>
<li><a href="COMPMUT_exp_data_6.csv"><code>COMPMUT_exp_data_6.csv</code></a>. Count data from medium-term cultures of different plasmid variants in soil. Data was collected by selective plating and replica plating. <code>replicate</code> indicates different transconjugants; <code>variant</code> is an abbreviated name referring to the population from which the clone came; <code>marker</code> indicates the antibiotic marker of the plasmid-bearing strain (Gm = plasmid initially in the gentamicin-resistant background; Sm = plasmid initially in the streptomycin-resistant <em>lacZ</em> background); <code>media</code> indicates the selective media (KB = no antibiotics, Gm3 = gentamicin at 3 µg/ml, Sm50 = streptomycin at 50 µg/ml); <code>dilution</code> indicates the dilution factor; <code>spread</code> indicates the volume spread in µl; <code>count_white</code> indicates the number of white (gentamicin-resistant) colonies; <code>count_blue</code> indicates the number of blue (streptomycin-resistant) colonies; <code>hg_kept</code> indicates the number of colonies from the originally-plasmid-free strain that had become mercury resistant; <code>hg_lost</code> indicates the number of colonies from the originally-plasmid-bearing strain that had become mercury sensitive.</li>
<li><a href="COMPMUT_exp_data_7.csv"><code>COMPMUT_exp_data_7.csv</code></a>. Growth curve data. <code>exp</code> refers to the plate in which the experiment was performed; <code>well</code> refers to plate well; <code>cycle</code> is the measurement cycle; <code>time</code> is time in seconds; <code>t_rep</code> is transformant replicate; <code>exp_rep</code> is experimental replicate; <code>pME6032</code> is the insert (if any); <code>IPTG</code> is IPTG concentration (0 µM or 100 µM); <code>OD600_corr</code> is the corrected OD<sub>600</sub> (OD<sub>600</sub> minus the mean blank well measurement).</li>
<li><a href="COMPMUT_exp_data_8.csv"><code>COMPMUT_exp_data_8.csv</code></a>. pQBR57 par segregation data. <code>code</code> is a shorthand for each experimental condition, <code>timepoint</code> is transfer number, <code>replicate</code> is independent transconjugant/transformant replicate, <code>par</code> is whether pUCP18 was empty or whether it carried pQBR57 par, <code>plasmid</code> is whether it carried pQBR57 or pQBR103, <code>spread</code> is volume of dilution spread on plate, <code>dilution</code> is (log10) dilution factor), <code>count</code> is total colonies counted, <code>count_hgs</code> is sensitive colonies as assessed by replica plating.</li>
<li><a href="COMPMUT_exp_data_9.csv"><code>COMPMUT_exp_data_9.csv</code></a>. Ectopic PFLU4242 expression in ∆gacS strains growth curve data. <code>well</code> is the well in the plate, <code>cycle</code> is measurement cycle (measurements taken every 15 mins), <code>replicate</code> is independent transformant/transconjugant replicate, <code>pME6032</code> is whether pME6032 was empty or whether it carried PFLU4242, <code>plasmid</code> is the pQBR plasmid carried, <code>IPTG</code> is whether or not cells were induced with IPTG, <code>OD600_corr</code> is the OD600 reading for that well, with blank values subtracted.</li>
</ul>
<h3 id="rnaseq-analysis">RNAseq analysis</h3>
<ul>
<li><a href="COMPMUT_RNAseq_1_de_table_chr_vanc.csv"><code>COMPMUT_RNAseq_1_de_table_chr_vanc.csv</code></a>. Differential expression of chromosomal genes relative to the plasmid-free, unameliorated ancestral strain, as determined by edgeR. Data is in ‘wide’ format, with rows referring to PFLU locus tags, and columns referring to each treatment. For each treatment, there is a column for log<sub>2</sub> fold-change (<code>logFC</code>), log<sub>2</sub> counts-per-million-per-gene (<code>logCPM</code>), test statistic for the quasi-likelihood F test (<code>F</code>), associated p-value (<code>PValue</code>), and Benjamini-Hochberg adjusted p-value (<code>FDR</code>).</li>
<li><a href="COMPMUT_RNAseq_2_de_table_chr_comp.csv"><code>COMPMUT_RNAseq_2_de_table_chr_vanc.csv</code></a>. Differential expression of chromosomal genes relative to the plasmid-bearing, unameliorated ancestral strain, as determined by edgeR. Columns are similar to <code>COMPMUT_RNAseq_1_de_table_chr_vanc.csv</code>.</li>
<li><a href="COMPMUT_RNAseq_3_de_table_pQ57.csv"><code>COMPMUT_RNAseq_3_de_table_pQ57.csv</code></a>. Differential expression of pQBR57 genes, relative to the unameliorated condition. Columns are similar to <code>COMPMUT_RNAseq_1_de_table_chr_vanc.csv</code>.</li>
<li><a href="COMPMUT_RNAseq_4_de_table_pQ103.csv"><code>COMPMUT_RNAseq_4_de_table_pQ103.csv</code></a>. Differential expression of pQBR103 genes, relative to the unameliorated condition. Columns are similar to <code>COMPMUT_RNAseq_1_de_table_chr_vanc.csv</code>.</li>
<li><a href="COMPMUT_RNAseq_5_tpm.csv"><code>COMPMUT_RNAseq_5_tpm.csv</code></a>. Table with values of mean transcripts-per-million (TPM) for each gene, averaged over replicates. Column <code>pc</code> refers to whether the gene is on the plasmid (<code>pla</code>) or chromsome (<code>chr</code>); <code>amelioration</code> refers to the amelioration; <code>plasmid</code> refers to the plasmid, <code>trt</code> is a code referring to the combination of plasmid and amelioration; <code>locus_tag</code> refers to the locus tag for the corresponding GenBank file; <code>group</code> is a combination of the <code>amelioration</code> and <code>plasmid</code> columns; <code>tpm</code> is mean transcripts-per-million across replicates; <code>n</code> gives number of replicates; <code>se</code> is standard error; <code>ci</code> is 95% confidence intervals; and <code>log2tpm</code> is a log<sub>2</sub> transformation of <code>tpm</code>.</li>
</ul>
<h3 id="compensatory-mutations-from-previous-studies">Compensatory mutations from previous studies</h3>
<ul>
<li><a href="COMPMUT_mutations.csv"><code>COMPMUT_mutations.csv</code></a>. Compensatory mutations aggregated from published studies. Note: data was not available for all columns for all mutations.</li>
</ul>
